### ISSUE TITLE
Column resize doesn't work on Android

### ISSUE THREAD

**Issue by davidhaimson** [schrodinger/fixed-data-table-2#277]
Our data grid is configured to allow column resizing. We set the `onColumnResizeEndCallback` on `Table`, and we set `touchScrollEnabled` to true. Using release 0.8.8, column resize works quite well on iOS, but very poorly on Android: on Android, it works about one time in ten.

## Expected Behavior
On Android, drag a column header right or left: the column should resize.

## Current Behavior
On Android, drag a column header right or left: sometimes nothing happens, sometimes dragging selects text, sometimes only the one-row drag handle appears (but dragging it does nothing), sometimes (about one time in ten) it actually resizes the column.

## Possible Solution
As a guess, this might work better if it handled pointer events instead of touch events on Android.

## Context
We would like to update to release 0.8.8, because we need column resize on touch devices, but it doesn't work on Android.

## Your Environment

* Version used: 0.8.8
* Browser Name and version: Chrome Version 63.0.3239.111
* Operating System and version (desktop or mobile): Android 7.1.1; Nexus 6P Build/N4F26O
* Link to your project: (not public)

**Comment by wcjordan** [comment]
Our mobile support is definitely very beta.  I'm happy to accept a PR on this and help point you to the areas of the code which may be useful if you'd like.

**Comment by davidhaimson** [comment]
>I'm happy to accept a PR on this

I'll see whether management will support me working on this.

**Comment by davidhaimson** [comment]
>>I'm happy to accept a PR on this
>
>I'll see whether management will support me working on this.

My guess is that if I make it use Pointer events where available, rather than Touch events, that will solve the problem. I've been given permission to try that, and if that fixes it, great, and if not, sorry.

Realistically, how much time do you think that will require?

**Comment by davidhaimson** [comment]
Also, it seems prudent for me to wait for your fix for https://github.com/schrodinger/fixed-data-table-2/issues/276#issuecomment-355673267.

**Comment by wcjordan** [comment]
Just pushed the fix for #276.  In theory, you just need to tweak the events here: 
https://github.com/schrodinger/fixed-data-table-2/blob/master/src/FixedDataTableCell.js#L248

**Comment by davidhaimson** [comment]
That does look like the starting place. Thanks!

**Comment by SirGandal** [comment]
@davidhaimson while testing #246 I found that sometimes it can be tricky to hit the target area for the resize handle as it is just a vertical line 4px wide. Have you tried making the target bigger to see if that _fixes_ it for android. It could be that iOS tries to be smart and helps you in hitting the target.
For testing purposes a simple rule like this might be enough:
`.fixedDataTableCellLayout_columnResizerKnob {
    width: 20px;
}`

**Comment by davidhaimson** [comment]
I'll give that a try, but I'm pretty sure that isn't the whole problem. I'm having pretty good success with using Pointer events. I did have to add the [Pointable](https://github.com/MilllerTime/react-pointable) component, since React doesn't support Pointer events.

**Comment by davidhaimson** [comment]
I've got a working version of this. It works correctly on:
- Chrome/Android
- Cordova/Android
- Safari/iOS
- Cordova/iOS
- Chrome/Mac
- Safari/Mac
- Firefox/Mac

It does not work on:
- IE11/Win7
- Edge/Win10

I think to fix it there I will need to add Pointer event handling to DOMMouseMoveTracker. Is there something special about `vendor_upstream`?

**Comment by wcjordan** [comment]
Ya, vendor_upstream is [fbjs|https://github.com/facebook/fbjs/tree/master/packages/fbjs] (from before fbjs existed).

It would be preferable to reach out on the fbjs repo and as if they'd accept a PR for it or how they recommend handling it.
DOMMouseMoveTracker can be found here:
https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/dom/DOMMouseMoveTracker.js

**Comment by davidhaimson** [comment]
I fixed it, and my team lead is reviewing it. I tested it and it works correctly on:
- Chrome/Android
- Cordova/Android
- Safari/iOS11
- Cordova/iOS11
- Chrome/Mac
- Safari/Mac
- Firefox/Mac
- IE11/Win7
- Edge/Win10

I managed to fix it with some very minor changes to FixedDataTableCell.js.

I did not need to handle Pointer events at all, so I did not end up adding Pointable.

I did not need to make any changes to DOMMouseMoveTracker, but I did find a bug in that component:
In `CaptureMouseMoves`:
```
      this._eventOutToken = EventListener.listen(
        this._domNode,
        'mouseout',
        this.onMouseEnd
      );
```
 Fortunately, `this.onMouseEnd` is a misspelling for `this._onMouseEnd`. If you handle the `mouseout` event here, it doesn't work at all. You want to ignore the `mouseout` event, q.v.
I hate having to rely on somebody else's bug!

I'll create a PR in your repo when we're satisfied with it.

**Comment by wcjordan** [comment]
Seems reasonable, I can review once it's created.  Could you also fix the onMouseEnd issue in DOMMouseMoveTracker?  It looks like that's forked enough from fbjs that we'll need to maintain it either way at this point.

**Comment by davidhaimson** [comment]
OK, fix it right there in vendor_upstream/dom/DOMMouseMoveTracker.js?

So far, I've done all my testing in our project. Is there some documentation for the testing paradigm for your project? Unit tests, for example? And I suppose there are other platforms you support.

**Comment by wcjordan** [comment]
Yep fixing it directly there is fine.

You can run the unit tests with `npm run test` (or yarn run test).  I also recommend trying out the examples which you can run with `npm run site-dev-server`.  

Unfortunately Facebook did not release their tests for FDT so they are sparse.  We've been adding more, but our efforts have been on the v1 beta branch so they're not available on the stable branch.

**Comment by sergek** [comment]
Hello, @wcjordan! The recent changes of the resize functionality in 0.8.8, 0.8.9 somehow did break previous resize workflow (in desktop browsers) and I'm trying to understand what should I change in my code to fix it.

Before those changes, it wasn't necessary to manipulate the isColumnResizing prop in any way by default, but I assume this is changed now.

Could you please elaborate a bit on how the proper onColumnResizeEndCallback should look like?
Should I just set isColumnResizing = false there and that's it, or there is more to be done?

**Comment by wcjordan** [comment]
@sergek could you see if #280 fixes the issue?  If it works on 0.8.7, but not #280, could you file as a separate issue with reproduction steps?

**Comment by davidhaimson** [comment]
In our app, we always set `isColumnResizing` to false. I don't even understand what that prop is for. fixed-data-table-2 knows when a column is resizing, and it lets the app know column resizing is done by calling `onColumnResizeEndCallback`. In the callback, we save the returned width in state, and use it to set the `width` prop on `Column`.

**Comment by wcjordan** [comment]
Fixed w/ release 0.8.10
